### PR TITLE
Add key alias functions and test

### PR DESCRIPTION
**Pull request by thu2thu2**

### CODE REVIEW

File: redstone/client.py [hunk 1]
```diff
@@ -605,6 +605,18 @@ def rotate(self, key_id, payload=None):
...
+        """Creates an alias for the specified key"""
+        resp = self.session.post("%s/api/v2/keys/%s/aliases/%s" % (self.endpoint_url, key_id, alias))
+        self._validate_resp(resp)
+        return resp.json()
+
+    def delete_key_alias(self, key_id: str, alias: str):
+        """Deletes an alias from the associated key"""
+        resp = self.session.delete("%s/api/v2/keys/%s/aliases/%s" % (self.endpoint_url, key_id, alias))
+        self._validate_resp(resp)
+        return resp.status_code
```
Comment (mrodden) [inline redstone/client.py:618]: Lets just return None on success. A failure case will raise an exception.

File: test/integration/test_keyprotect.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                                   region="us-south",
+                                   service_instance_id=self.instance_id,
+                                   )
+
+    def tearDown(self):
+        for key in self.kp.keys():
+            self.kp.delete(key.get('id'))
+
+        self.rc.delete_instance(self.instance_id)
+        self.kp.session.close()
```
Comment (mrodden) [inline test/integration/test_keyprotect.py:35]: You will probably want to catch the errors if one of the keys fails to delete, or if the instance fails to delete.

I would probably delete all the keys in a try block, and then in a finally clause try the delete on the instance.

**Review by mrodden — CHANGES_REQUESTED** [review]

File: redstone/client.py (hunk 1 above)
Comment (thu2thu2) [reply redstone/client.py:618]: Updated

File: test/integration/test_keyprotect.py (hunk 2 above)
Comment (thu2thu2) [reply test/integration/test_keyprotect.py]: Updated, thanks!

File: redstone/client.py [hunk 3]
```diff
@@ -645,6 +645,32 @@ def enable_key(self, key_id: str):
         )
         self._validate_resp(resp)
 
+    def create_key_alias(self, key_id: str, alias: str):
+        """
+        Creates an alias name for a key.
+        An alias name acts as an identifier just like key ID
```
Comment (mrodden) [inline redstone/client.py:651]:
```suggestion
        An alias is a user defined string that can be used in place of a normal UUID Key ID
```

**Review by mrodden — APPROVED** [review]

**Comment by mrodden** [comment]
Merged with 9ed7549

**Pull request closed without merging**